.github/workflows/unit-test.yml

```diff
@@ -33,27 +33,23 @@ jobs:
       - run: npm install -g npm@"<10.0.0"
         if: ${{ matrix.node_version == '14' || matrix.node_version == '16' }}
 
-      - name: restore lerna
+      - name: restore node_modules
         id: cache
         uses: actions/cache@v3
         with:
           path: |
             node_modules
-            package-lock.json
+            api/node_modules
             packages/*/node_modules
-            packages/*/package-lock.json
-            experimental/packages/*/node_modules
-            experimental/packages/*/package-lock.json
-          key: node-tests-${{ runner.os }}-${{ matrix.node_version }}-${{ hashFiles('package.json', 'packages/*/package.json', 'experimental/packages/*/package.json') }}-04292022
-
-      # temporary fix for https://github.com/nrwl/nx/issues/19022
-      - run: npm install --save-dev nx@16.7.4 @nx/devkit@16.7.4
-        if: ${{ matrix.node_version == '14' }}
+            experimental/*/*/node_modules
+            integration-tests/*/node_modules
+            selenium-tests/node_modules
+            examples/*/node_modules
+          key: node-tests-${{ runner.os }}-${{ matrix.node_version }}-${{ hashFiles('package-lock.json') }}-10302023
```

Comment:
I actually think the cache is causing tests to pass when they otherwise wouldn't. Locally I get dependency errors from a clean build on this branch.

Reply:
would you mind sharing a piece of error you encountered locally?

Reply:
Yes. First when running `npm run compile` it fails because `rimraf` is not found. 

```
npm run compile

> opentelemetry@0.1.0 precompile /Users/daniel.dyla/otel/opentelemetry-js
> lerna run version && npm run submodule

internal/modules/cjs/loader.js:934
  throw err;
  ^

Error: Cannot find module 'rimraf'
Require stack:
- /Users/daniel.dyla/otel/opentelemetry-js/node_modules/@lerna/legacy-package-management/dist/index.js
- /Users/daniel.dyla/otel/opentelemetry-js/node_modules/lerna/dist/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:931:15)
    at Function.Module._load (internal/modules/cjs/loader.js:774:27)
    at Module.require (internal/modules/cjs/loader.js:1003:19)
    at require (internal/modules/cjs/helpers.js:107:18)
    at libs/core/src/lib/rimraf-dir.ts (/Users/daniel.dyla/otel/opentelemetry-js/node_modules/@lerna/legacy-package-management/dist/index.js:3587:29)
    at __init (/Users/daniel.dyla/otel/opentelemetry-js/node_modules/@lerna/legacy-package-management/dist/index.js:10:56)
    at libs/core/src/index.ts (/Users/daniel.dyla/otel/opentelemetry-js/node_modules/@lerna/legacy-package-management/dist/index.js:3955:5)
    at __init (/Users/daniel.dyla/otel/opentelemetry-js/node_modules/@lerna/legacy-package-management/dist/index.js:10:56)
    at libs/commands/add/src/command.ts (/Users/daniel.dyla/otel/opentelemetry-js/node_modules/@lerna/legacy-package-management/dist/index.js:4626:5)
    at __require (/Users/daniel.dyla/otel/opentelemetry-js/node_modules/@lerna/legacy-package-management/dist/index.js:13:50) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/daniel.dyla/otel/opentelemetry-js/node_modules/@lerna/legacy-package-management/dist/index.js',
    '/Users/daniel.dyla/otel/opentelemetry-js/node_modules/lerna/dist/cli.js'
  ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! opentelemetry@0.1.0 precompile: `lerna run version && npm run submodule`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the opentelemetry@0.1.0 precompile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/daniel.dyla/.npm/_logs/2023-11-01T12_59_57_666Z-debug.log
```

After installing rimraf with `npm i -D rimraf` in the root, it continues until the `submodule` script tries to run `buf` and fails again with a similar issue:

```
>  Lerna (powered by Nx)   Successfully ran target version for 41 projects (17s)
 

> opentelemetry@0.1.0 submodule /Users/daniel.dyla/otel/opentelemetry-js
> git submodule sync --recursive && git submodule update --init --recursive

Synchronizing submodule url for 'experimental/packages/otlp-grpc-exporter-base/protos'
Synchronizing submodule url for 'experimental/packages/otlp-proto-exporter-base/protos'

> opentelemetry@0.1.0 compile /Users/daniel.dyla/otel/opentelemetry-js
> lerna run protos:generate && tsc --build tsconfig.json tsconfig.esm.json tsconfig.esnext.json

lerna notice cli v6.6.2
lerna info versioning independent


    ✖  @opentelemetry/instrumentation-grpc:protos:generate
       > @opentelemetry/instrumentation-grpc@0.44.0 protos:generate /Users/daniel.dyla/otel/opentelemetry-js/experimental/packages/opentelemetry-instrumentation-grpc
       > cd test/fixtures && buf generate
       
       sh: buf: command not found
       npm ERR! code ELIFECYCLE
       npm ERR! syscall spawn
       npm ERR! file sh
       npm ERR! errno ENOENT
       npm ERR! @opentelemetry/instrumentation-grpc@0.44.0 protos:generate: `cd test/fixtures && buf generate`
       npm ERR! spawn ENOENT
       npm ERR! 
       npm ERR! Failed at the @opentelemetry/instrumentation-grpc@0.44.0 protos:generate script.
       npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
       npm WARN Local package.json exists, but node_modules missing, did you mean to install?
       
       npm ERR! A complete log of this run can be found in:
       npm ERR!     /Users/daniel.dyla/.npm/_logs/2023-11-01T13_02_53_337Z-debug.log
       
    ✔  @opentelemetry/otlp-grpc-exporter-base:protos:generate (12s)
    ✔  @opentelemetry/otlp-proto-exporter-base:protos:generate (12s)

 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  Lerna (powered by Nx)   Ran target protos:generate for 3 projects (12s)
 
    ✔    2/3 succeeded [0 read from cache]
 
    ✖    1/3 targets failed, including the following:
         - @opentelemetry/instrumentation-grpc:protos:generate
 
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! opentelemetry@0.1.0 compile: `lerna run protos:generate && tsc --build tsconfig.json tsconfig.esm.json tsconfig.esnext.json`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the opentelemetry@0.1.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/daniel.dyla/.npm/_logs/2023-11-01T13_03_04_036Z-debug.log
```

Reply:
Actually I think I discovered my error. I hid the comments so they don't distract from the rest of this thread. I still think caching is not needed with `npm ci` because the cached files are removed anyway.

Reply:
You are right, https://docs.npmjs.com/cli/v10/commands/npm-ci mentioned that npm-ci is going to remove the whole node_modules to do a clean install. Caches removed.

Reply:
You can however [enable npm cache in actions/setup-node](https://github.com/actions/setup-node#caching-global-packages-data) which will cache the local packages cache ([npm-cache](https://docs.npmjs.com/cli/v10/commands/npm-cache))